engine/src/main/java/io/camunda/zeebe/engine/processing/dmn/EvaluateDecisionProcessor.java

```diff
@@ -0,0 +1,118 @@
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.dmn;
+
+import io.camunda.zeebe.engine.processing.common.DecisionBehavior;
+import io.camunda.zeebe.engine.processing.common.Failure;
+import io.camunda.zeebe.engine.processing.streamprocessor.CommandProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedRejectionWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.deployment.PersistedDecision;
+import io.camunda.zeebe.protocol.impl.record.value.decision.DecisionEvaluationRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.DecisionEvaluationIntent;
+import io.camunda.zeebe.protocol.record.value.ErrorType;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+import io.camunda.zeebe.stream.api.state.KeyGenerator;
+import io.camunda.zeebe.util.Either;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import io.camunda.zeebe.util.collection.Tuple;
+
+public class EvaluateDecisionProcessor implements CommandProcessor<DecisionEvaluationRecord> {
+
+  private static final String ERROR_MESSAGE_NO_IDENTIFIER_SPECIFIED =
+      "Expected either a decisionId or a decisionKey greater than -1, but none or both provided";
+
+  private final DecisionBehavior decisionBehavior;
+  private final TypedRejectionWriter rejectionWriter;
+  private final TypedResponseWriter responseWriter;
+  private final StateWriter stateWriter;
+  private final KeyGenerator keyGenerator;
+  private boolean success;
+
+  public EvaluateDecisionProcessor(
+      final DecisionBehavior decisionBehavior,
+      final KeyGenerator keyGenerator,
+      final Writers writers) {
+
+    this.decisionBehavior = decisionBehavior;
+    this.keyGenerator = keyGenerator;
+    stateWriter = writers.state();
+    rejectionWriter = writers.rejection();
+    responseWriter = writers.response();
+  }
+
+  @Override
+  public boolean onCommand(
+      final TypedRecord<DecisionEvaluationRecord> command,
+      final CommandControl<DecisionEvaluationRecord> commandControl) {
+
+    final DecisionEvaluationRecord record = command.getValue();
+    final Either<Failure, PersistedDecision> decisionOrFailure = getDecision(record);
+
+    decisionOrFailure
+        .flatMap(persistedDecision -> decisionBehavior.findAndParseDrgByDecision(persistedDecision))
+        .ifRightOrLeft(
+            drg -> {
+              final var decision = decisionOrFailure.get();
+              final var variables = record.getVariablesBuffer();
+              final var evaluationResult =
+                  decisionBehavior.evaluateDecisionInDrg(
+                      drg, BufferUtil.bufferAsString(decision.getDecisionId()), variables);
```

Comment:
❓ What is the intended behavior if the evaluation fails? Do we want to send a command rejection with the failure message? If so we need to add a check here.

Or do we want to send a success response with the failure in it? I think the first option is nicer.

Reply:
### So the two cases are
1. A command with invalid arguments is sent:
    * The command is rejected since the engine can't process it.
2. A command with valid arguments is sent. The `DecisionEngine` is used to evaluate the decision:
    1. The evaluation is successful
         1. We increment the metrics
         2. We write an `EVALUATED` record
         3. We respond with the result
     2. The evaluation fails
         1. We increment the metrics.
         2. **We write a `FAILED` record** or **We reject the command**
         3. We respond with the failure. 

### To view it from the engine's perspective
If a command is processed, it needs to append an event, so it can be replayed (this might not matter so much for standalone decision evaluation). Since we increment the metrics when we use the `DecisionEngine` no matter the result, that indicates we accepted the command, so we need to write an event. 

Does it make sense to reject the command then?

### From a users perspective
As a user, the response differs depending on the input. If they provide bad input arguments, it's a rejection. If they didn't provide the correct variables, it's a successful response but with a failure. This makes sense. 

For example, if a user tries to start a process instance, but provides the wrong BPMN ID or KEY, they should receive a rejection. But if they successfully create a Process Instance, but failed to provide a variable, they might receive an Incident. So a different behavior is valid depending on the user's input.

Note: There might be a similar behavior with the [JobFailProcessor](https://github.com/camunda/zeebe/blob/f549e5521664277517abb20f369f17edf47e2eb5/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobFailProcessor.java#L131).

**So the second option might make more sense. WDYT @remcowesterhoud ?**

Reply:
I updated #1516 so we don't forget it.

Reply:
You linked the wrong issue, but I'll trust you wrote it down 😄